library/src/main/java/com/novoda/downloadmanager/DownloadBatchRequirementRules.java

```diff
@@ -21,12 +23,27 @@ void addRule(DownloadBatchRequirementRule storageRequirementRule) {
 
     @Override
     public boolean hasViolatedRule(DownloadBatchStatus downloadBatchStatus) {
```

Comment:
this method won't be used at all, so maybe we should just remove it?

Reply:
But then a "composite rule" could no longer inherit from "rule". We'd break the chain, which I think is more intrusive in regard to the original code.

And not necessarily it won't be used. There may be cases where calling code could only be interested whether *any* rule has been violated, without needing to know which one. 

It doesn't even have to know if the rule is composite or not (and `hasViolatedRule` is the lowest common denominator for rules).